pkg/platform/kube/functionres/lazy.go

```diff
@@ -2184,23 +2191,78 @@ func (lc *lazyClient) resolveFailFast(pods []v1.Pod,
 				}
 			}
 		}
+	}
 
+	// check each pod's conditions to determine if there is at least one unschedulable pod
+	errGroup, _ := errgroup.WithContext(context.TODO(), lc.logger)
+	lock := sync.Mutex{}
+	scaleUpOccurred := false
+	for _, pod := range pods {
+		pod := pod
 		for _, condition := range pod.Status.Conditions {
 
 			// check if the pod is in pending state, and the reason is that it is unschedulable
 			// (meaning no k8s node can currently run it, because of insufficient resources etc..)
 			if pod.Status.Phase == v1.PodPending && condition.Reason == "Unschedulable" {
 
-				// TODO: if pods has an event that cluster is going to scale up - don't return error
-
-				return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+				errGroup.Go("WaitAndCheckAutoScaleEvents", func() error {
+					time.Sleep(15 * time.Second)
+
+					// check if the pod is unschedulable due to scaling up
+					triggeredScaleUp, err := lc.isPodAutoScaledUp(pod)
+					if err != nil {
+						return errors.Wrapf(err,
+							"Failed to resolve pod (%s) triggered a node scale up",
+							pod.Name)
+					}
+					if triggeredScaleUp {
+						lc.logger.InfoWith("Nuclio function pod has triggered a scale up", "podName", pod.Name)
+						lock.Lock()
+						defer lock.Unlock()
+
+						scaleUpOccurred = true
+					} else {
+						return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+					}
+					return nil
+				})
 			}
 		}
 	}
+	if err := errGroup.Wait(); err != nil {
+		return errors.Wrap(err, "Failed to verify at least one pod schedulability")
+	}
+	if scaleUpOccurred {
 
+		// TODO: Increase wait timeout?
+		lc.logger.Debug("Scale up occurred")
+	}
 	return nil
 }
 
+func (lc *lazyClient) isPodAutoScaledUp(pod v1.Pod) (bool, error) {
+
+	// get pod events to check if pod triggered auto scale
+	podEvents, err := lc.kubeClientSet.CoreV1().Events(pod.Namespace).List(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("involvedObject.name=%s", pod.Name),
+	})
+	if err != nil {
+		return false, errors.Wrap(err, "Failed to list pod events")
+	}
+
+	for _, event := range podEvents.Items {
+
+		if event.Source.Component == "cluster-autoscaler" {
+			if event.Reason == "TriggerScaleUp" {
+				return true, nil
+			} else if event.Reason == "NotTriggerScaleUp" || event.Reason == "ScaleDown" {
```

Comment:
imo `"NotTriggerScaleUp"` is more interesting  here. because that means that the pod requested more resources than allowed in node group. log something about it